GOVERNANCE.md

```diff
@@ -61,19 +61,67 @@ maintainer has already discussed the candidacy with the candidate and a
+handful of other repositories, are all **core** and required pieces which culminate
+in officially supported and maintained releases of containerd.
+
+Non-core projects have strong affiliation with the containerd runtime, but
+operate in some ways similar to the traditional `contrib/` directory in many
+open source projects. They are most likely separately maintained, and potentially
+have a different release cycle and support guarantee than the main containerd project
+and its core subprojects. They may be written in other languages and therefore
+require different skills, developer tools, and CI systems than the core projects.
+For these reasons, non-core subprojects will be treated differently than core
+projects within the governance of the containerd organization.
+
+Both core and non-core subprojects must adhere to the CNCF
+[charter](https://www.cncf.io/about/charter/) and mission.
+
+### Adding core subprojects
+
+New core subprojects can request to be added to the containerd GitHub
+organization by submitting a project proposal via public forum (a
+`containerd/project` GitHub issue is the easiest way to provide this proposal).
+The existing maintainers are given five business days to discuss the new
 project, raise objections and cast their vote. Projects must be approved by at
 least 66% of the current maintainers by adding their vote.
 
 If a project is approved, a maintainer will add the project to the containerd
 GitHub organization, and make an announcement on a public forum.
 
+To make this transition to being a core subproject easier, please prepare your
+repository with the DCO signoff checks, the fileheader check, and by adding the
+appropriate Apache 2 license file to your project root as well as the boilerplate
+header to all applicable source files.
+
+### Adding non-core subprojects
+
+Non-core subprojects will also submit a project proposal via public forum, and
+should state that the project is expected to be **non-core**.
+
+The proposal should include a proposed list of maintainers who will manage
+the non-core project and provide general information on support, releases,
+stability, and any additional detail useful for the containerd maintainers to
+understand the scope and nature of the project.
+
+The existing maintainers are given five business days to discuss the new
+project, raise objections and cast their vote. Projects must be approved by at
+least 50% of the current maintainers by adding their vote.
+
+If a project is approved, a maintainer will add the project to the containerd
+GitHub organization and provide write access for that repository to the proposed
+maintainer list, as well as make an announcement on a public forum.
+
+Unlike core projects, non-core project maintainers can become reviewers and
+maintainers per containerd governance steps noted in this document, but are not
+automatically so by nature of maintaining a non-core subproject.
```

Comment:
Does this imply that once a project becomes `core`, all maintainers become containerd maintainers?

Reply:
I would say yes from how things work within containerd but unlike incubator projects in kube, these non-core projects don't really have a way into "core" status as they are either out of scope of the core or it just doesn't fit.  

I don't see non-core as a stepping point to core projects, you are either a core project that the runtime depends on and releases or your project is non-core.

Reply:
core projects are pretty rare going forward I would expect, but your statement is true for all prior core projects. When "cri" was moved in, we added all your CRI project maintainers to the list of all core containerd organization maintainers. All other repos that I can remember being contributed had full overlap (the person contributing was also a core maintainer) with the existing set of maintainers. If we think a project is of core importance to the release and support process of containerd runtime and plugins that are built-in, then I would think we would see the maintainers of that project as being integral to the overall project.

Reply:
> I don't see non-core as a stepping point to core projects, you are either a core project that the runtime depends on and releases or your project is non-core.

I see. That explains. :)

> If we think a project is of core importance to the release and support process of containerd runtime and plugins that are built-in, then I would think we would see the maintainers of that project as being integral to the overall project.

SG. Thanks for the clarification!